all-actors/src/main/java/org/sunbird/notification/actor/NotificationActor.java

```diff
@@ -47,9 +52,18 @@ public void notify(Request request) throws BaseException {
       NotificationValidator.validate(notificationRequest);
     }
     Map<String, Object> requestMap = request.getRequest();
-    List<FCMResponse> responses = Dispatcher.dispatch(requestMap, false);
+    List<FCMResponse> responses = new ArrayList<FCMResponse>();
     Response response = new Response();
-    response.getResult().put("response", responses);
+    if (System.getenv(SUNBIRD_NOTIFICATION_DEFAULT_DISPATCH_MODE) != null
+        && !SUNBIRD_NOTIFICATION_DEFAULT_DISPATCH_MODE_VAL.equalsIgnoreCase(
+            System.getenv(SUNBIRD_NOTIFICATION_DEFAULT_DISPATCH_MODE))) {
+      responses = Dispatcher.dispatch(requestMap, false);
```

Comment:
To keep this simple, an alternative could be this...
response = Dispatcher.dispatch(requestMap);

Internally within the dispatch function, read the env value and do sync/async operation. Leave the responsibility and knowledge at the right layers.

Reply:
done.